### PR TITLE
Added `--namespaces` flag to `glooctl check`

### PR DESCRIPTION
**Pull request by Bslabe123**
# Description

Added a —namespaces flag that signals to `glooctl` check resources on a over an array of namespaces. 

# Context

The usage is as follows: `glooctl check —namespaces gloo-system,gloo-portal,foo,bar`. This flag reuses the functionality of the -n flag and iterates until failing. This flag also does not de-duplicate. The `--namespaces` flag has a higher priority than the `-n` flag.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1299

### CODE REVIEW

File: docs/content/reference/cli/glooctl_check.md [hunk 1]
```diff
@@ -17,10 +17,11 @@ glooctl check [flags]
 ### Options
 
 ```
-  -x, --exclude strings     check to exclude: (deployments, pods, upstreams, upstreamgroup, auth-configs, rate-limit-configs, secrets, virtual-services, gateways, proxies, xds-metrics)
-  -h, --help                help for check
-  -n, --namespace string    namespace for reading or writing resources (default "gloo-system")
-  -o, --output OutputType   output format: (json, table) (default table)
+  -x, --exclude strings      check to exclude: (deployments, pods, upstreams, upstreamgroup, auth-configs, rate-limit-configs, secrets, virtual-services, gateways, proxies, xds-metrics)
+  -h, --help                 help for check
+  -n, --namespace string     namespace for reading or writing resources (default "gloo-system")
```
Comment (nfuden) [inline docs/content/reference/cli/glooctl_check.md:22]: glooctl check doesnt write anything? so this seems wrong?

File: changelog/v1.13.0-beta26/glooctl-namespaces-flag.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: NEW_FEATURE
+    issueLink: https://github.com/solo-io/gloo/issues/1299
```
Comment (nfuden) [inline changelog/v1.13.0-beta26/glooctl-namespaces-flag.yaml:3]: Should add resolves false as this is a good change but does not resolve the issue as stated. In particular the requesting feature was both for namespaces and for sub pods by a label

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 3]
```diff
@@ -64,7 +64,21 @@ func RootCmd(opts *options.Options, optionsFunc ...cliutils.OptionsFunc) *cobra.
 
 			printer = printers.P{OutputType: opts.Top.Output}
 			printer.CheckResult = printer.NewCheckResult()
-			err := CheckResources(opts)
+
+			var err error
+			if len(opts.Check.Namespaces) != 0 {
```
Comment (nfuden) [inline projects/gloo/cli/pkg/cmd/check/root.go:69]: if len of namespaces is 0 cant we just add gloo-system to the list? Would make the actual called code safer

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 4]
```diff
@@ -64,7 +64,21 @@ func RootCmd(opts *options.Options, optionsFunc ...cliutils.OptionsFunc) *cobra.
...
+					printer.AppendMessage("Checking " + namespace + " namespace.\n")
+					opts.Metadata.Namespace = namespace
+					err = CheckResources(opts)
+					if err != nil {
+						return err
+					}
+				}
+			} else {
+				printer.AppendMessage("Checking default namespace (gloo-system).\n")
+				err = CheckResources(opts)
```
Comment (jenshu) [inline projects/gloo/cli/pkg/cmd/check/root.go:80]: doesn't CheckResources still [check some resources in all namespaces](https://github.com/solo-io/gloo/blob/c83ce773dd0437c7d52fa70874608699c136b08c/projects/gloo/cli/pkg/cmd/check/root.go#L125)?

File: projects/gloo/cli/pkg/cmd/check/root.go (hunk 3 above)
Comment (Bslabe123) [reply projects/gloo/cli/pkg/cmd/check/root.go:69]: Think I have an appropriate fix

File: docs/content/reference/cli/glooctl_check.md (hunk 1 above)
Comment (Bslabe123) [reply docs/content/reference/cli/glooctl_check.md:22]: line 22 was unedited only the spacing was changed, should we change the description?

File: projects/gloo/cli/pkg/cmd/check/root.go (hunk 4 above)
Comment (Bslabe123) [reply projects/gloo/cli/pkg/cmd/check/root.go]: Changed my approach, the --namespace flag now sets the namespaces we're checking for all resources

File: docs/content/reference/cli/glooctl_check.md [hunk 5]
```diff
@@ -17,10 +17,11 @@ glooctl check [flags]
...
+      --namespaces strings   namespaces for reading resources
```
Comment (kdorosh) [inline docs/content/reference/cli/glooctl_check.md:23]: should we instead consider a label selector like kubectl does for namespace selection?

Comment (kdorosh) [reply docs/content/reference/cli/glooctl_check.md:23]: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/

Comment (kdorosh) [reply docs/content/reference/cli/glooctl_check.md:23]: seems more expressive and powerful and you can still do list based requirements via a set (... in [ ] )

we just do `kubectl get ns -l "passthrough"` to get namespaces and then use that list to get all other resources

**Comment by Bslabe123** [comment]
New PR https://github.com/solo-io/gloo/pull/7402

**Pull request closed without merging**